index.js

```diff
@@ -762,6 +765,14 @@
     typeEq('sanctuary-def/Type')
   );
 
+  var VariadicType = NullaryType(
+    'sanctuary-def/VariadicType',
+    '',
+    function(x){
+      return test([], Type, x) || test([], AnyFunction, x);
```

Comment:
I tried different ways of testing:

1. `test([], Type, applyParameterizedType(x))` - This blows up if someone would provide, say `[x => true]` as `env`.
1. `test([], Type, x) || test([], Function([Type, Type]), x) || test([], Function([Type, Type, Type]), x)` - This effectively behaves the same as `AnyFunction`, because `test` does not apply `x` to determine its types.
1. This left me with the method seen above.

The decision, really, is between `1` and `3`, where `3` is overly permissive, and `1` blows up with an ugly error. Maybe to get the best of both worlds, I should combine them as:

```js
test([], Type, x) || test([], AnyFunction, x) && test([], Type, applyParameterizedType(x))
```